File: src/gestorAplicacion/enrutadorHFC/Conexion.java
package gestorAplicacion.enrutadorHFC;

import gestorAplicacion.host.Cliente;

public class Conexion extends Router{

	//ESTA CLASE ES FUNDAMENTAL EN LAS FUNCIONALIDADES DEL TEST Y REPORTE PARA EL CALCULO DE LAS INTENSIDADES DE FLUJO---DE ELLA SE OBTIENEN LAS VELOCIDADES EN "TIEMPO REAL"

	//ATRIBUTOS
	private Cliente cliente;

	//CONSTRUCTORES
	public Conexion(String ip, int up, int down, boolean online, int generacion, Cliente cliente, Servidor servidor) {
		super(up, down, online, servidor);
	    this.cliente=cliente;
	}
  
	// SETTERS Y GETTERS

	//METODO HEREDADO DE ROUTER SOBREESCRITO CON CUAL SE APLICA LIGADURA DINÁMICA
	public int getVelocidad() {
    	int velocidadActual = ((cliente.getModem().actualizarVelocidad(cliente).get(0) + cliente.getModem().actualizarVelocidad(cliente).get(1)) / 2)*generacion*generacion;
		return velocidadActual;
	}

	public Cliente getCliente() {
		return cliente;
	}

	public void setCliente(Cliente cliente) {
		this.cliente = cliente;
	}

}

File: Python/src/gestorAplicacion/enrutadorHFC/conexion.py
from router import Router

class Conexion(Router):

  # //ESTA CLASE ES FUNDAMENTAL EN LAS FUNCIONALIDADES DEL TEST Y REPORTE PARA EL CALCULO DE LAS INTENSIDADES DE FLUJO---DE ELLA SE OBTIENEN LAS VELOCIDADES EN "TIEMPO REAL"

  # CONSTRUCTORES
  def __init__(self,ip,up,down,online,generacion,cliente,servidor):

    # ATRIBUTOS
    self._cliente=cliente
    super().__init__(up,down,online,servidor)

  # // SETTERS Y GETTERS

  # //METODO HEREDADO DE ROUTER SOBREESCRITO CON EL CUAL SE APLICA "LIGADURA DINÁMICA"
  def getVelocidad(self):
    velocidadActual=(((self._cliente.getModem().actualizarVelocidad(self._cliente)[0] + self._cliente.getModem().actualizarVelocidad(self._cliente)[0] )/2)*self.generacion*self.generacion) 
    return velocidadActual
  
  def getCliente(self):
    return self._cliente
  
  def setCliente(self,cliente):
    self._cliente=cliente